Classes/EventStore/AbstractEventSourcedRepository.php

```diff
@@ -88,15 +85,9 @@ public function __construct()
     final public function save(EventSourcedAggregateRootInterface $aggregate, int $expectedVersion = null)
     {
         $streamName = $this->streamNameResolver->getStreamNameForAggregate($aggregate);
-        $uncommittedEvents = $aggregate->pullUncommittedEvents();
-        $version = $aggregate->getReconstitutionVersion() + 1;
-        $uncommittedEventsWithMetadata = array_map(function (EventInterface $event) use (&$version) {
-            return new EventWithMetadata($event, new EventMetadata([EventMetadata::VERSION => $version ++]));
```

Comment:
Well, the issue is that we basically removed the whole concept of aggregate version, which is normally used to guarantee replaying aggregates with correctly ordered events, when we moved that here. It was no longer an aggregate version, but a publishing-version at most (which is rather useless).

The aggregate typically records (in the event metadata) in which order it applied the events in the first place, so that when reconstituting it, it will deterministically result in the same aggregate state.
The question is, does our environment (share-nothing single process per request) somehow influence that requirement such that the deterministic reconstitution is guaranteed anyway? I can't 100% say yes currently.